### PR TITLE
Handle avro generator in gradle

### PR DESCRIPTION
**Pull request by glefloch**
I saw that avro code generator was integrated in the main branch. 
This add support in the gradle plugin.

### CODE REVIEW

**Comment by geoand** [comment]
Seems like you need to:

```
Error: Dependencies in integration-tests/gradle/pom.xml are outdated!
Error: Run update-dependencies.sh in integration-tests/gradle and add the modified pom.xml file to your commit.
Error: Process completed with exit code 1.
```

:)

**Comment by glefloch** [comment]
right! I tried to do it manually but it didn't work :) I will run the script.

**Comment by glefloch** [comment]
@famod I don't understand what is going on.. I ran the script on my machine (both new version working on mac os, and current version in a container) and the pom does not change but ci is complaining.. have you an idea?

**Comment by famod** [comment]
@glefloch I'll have a look!

**Comment by famod** [comment]
I added `git diff` to the workflow step (something I should have done a while ago):
```diff
diff --git a/integration-tests/gradle/pom.xml b/integration-tests/gradle/pom.xml
index bfd5a86c8e..e5ab1bcb4a 100644
--- a/integration-tests/gradle/pom.xml
+++ b/integration-tests/gradle/pom.xml
@@ -94,21 +94,11 @@
             <artifactId>quarkus-resteasy</artifactId>
             <version>${project.version}</version>
         </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-resteasy-jsonb</artifactId>
-            <version>${project.version}</version>
-        </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-test-devtools</artifactId>
             <version>${project.version}</version>
         </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-vertx</artifactId>
-            <version>${project.version}</version>
-        </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-arc-deployment</artifactId>
@@ -200,32 +190,6 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-resteasy-jsonb-deployment</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-            <scope>test</scope>
-            <exclusions>
-                <exclusion>
-                    <groupId>*</groupId>
-                    <artifactId>*</artifactId>
-                </exclusion>
-            </exclusions>
-        </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-vertx-deployment</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-            <scope>test</scope>
-            <exclusions>
-                <exclusion>
-                    <groupId>*</groupId>
-                    <artifactId>*</artifactId>
-                </exclusion>
-            </exclusions>
-        </dependency>
         <!-- END update-dependencies.sh -->
     </dependencies>
```
So in CI, the script is removing vert.x and jsonb... Does that make any sense?

**Comment by famod** [comment]
Hm, looking at what you added/changed in this PR, I have to say I understand why the script is removing those dependencies: neither of the two deps was added to any of the `build*.gradle` files.

**Comment by glefloch** [comment]
Well I think it make sense. I tried reverting all modification in the pom running the script again but, I have the same result. I will drop those new dependency manually

**Comment by famod** [comment]
...and now I can also replicate the behavior locally. First the script didn't change anything for me locally either because I called it via ` integration-tests/gradle/update-dependencies.sh` but that will only work after #15823 was merged!
Currently you have to `cd integration-tests/gradle && ./update-dependencies.sh`.

**Comment by glefloch** [comment]
I'm running it in `integration-tests/gradle`

**Comment by famod** [comment]
Then you should see the same effect as in CI.

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
     public static final String QUARKUS_GENERATED_SOURCES = "quarkus-generated-sources";
     public static final String QUARKUS_TEST_GENERATED_SOURCES = "quarkus-test-generated-sources";
     // TODO dynamically load generation provider, or make them write code directly in quarkus-generated-sources
-    public static final String[] CODE_GENERATION_PROVIDER = new String[] { "grpc" };
+    public static final String[] CODE_GENERATION_PROVIDER = new String[] { "grpc", "avpr", "avsc" };
```
Comment (michalszynkiewicz) [inline devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusGenerateCode.java:35]: it's really bad that we need this list. Can we access classpath resources in here?

**Review by michalszynkiewicz — APPROVED** [review]

**Pull request merged**